---
title: "Bioinformatic analysis of genes"
author: "B"
date: "February 17, 2015"
output:
  html_document
---


## Reading data from the file and making data frame 

```{r, eval=FALSE}
load(file="retro-matrix-1-ips-vs-phdf.RData")
rownames(retroMatrix) <- retroMatrix[,1]
retroMatrix <- retroMatrix[,-c(1,8)]
```

## Transformation data frame to count data set

```{r, eval=FALSE}
library(DESeq)
cds <- newCountDataSet(retroMatrix, substr(colnames(retroMatrix), 1, 1))
```

## Normalization and estimating dispersion

```{r, eval=FALSE}
cds <- estimateSizeFactors(cds)
sizeFactors(cds) 
# Result of normlization for the first five columns
head(counts(cds[,1:5], normalized=TRUE))

cds <- estimateDispersions(cds)
# Estiamted and fitted values might be read from:
str(fitInfo(cds))

# Value of dispersion for every gene
head(fData(cds))

# Plot of estimated dispersion
plotDispEsts(cds)
```

## Comparision between two experimental conditions

```{r, eval=FALSE}
#Test of differential expression
res <- nbinomTest(cds, "P", "I")
head(res)

plotMA(res)
```

```{r, eval=FALSE}
# Selecting genes with p-value adjusted < 0.1
resSig = res[res$padj < 0.1,]

# Order selected genes by increasing p-value and decreasing fold change 
ordered <- resSig[order(resSig$pval, -resSig$foldChange), ]
head(ordered)

```

## Heatmap

Heatmap is drawn for the 50 most highly expressed genes.

```{r, eval=FALSE}
library(gplots)
library(RColorBrewer)
cds <- estimateDispersions(cds, method="blind")
select <- order(rowMeans(counts(cds)), decreasing=TRUE)[1:50]
hmcol <- colorRampPalette(brewer.pal(9, "GnBu"))(100)
heatmap.2(counts(cds)[select,], trace="none", col=hmcol, margin=c(10,8))
```
